# Kepler's laws of Planetary Motion!
## Let's do some exercises from Chapter 3 of the Astronomy book!
### Kepler's First Two Laws of Planetary Motion
The path of an object through space is called its orbit. Kepler initially assumed that the orbits of planets were circles, but doing so did not allow him to find orbits that were consistent with Brahe’s observations. Working with the data for Mars, he eventually discovered that the orbit of that planet had the shape of a somewhat flattened circle, or ellipse. Next to the circle, the ellipse is the simplest kind of closed curve, belonging to a family of curves known as conic sections

![conics](conics.png)


### Kepler's First Law
Each planet moves around the Sun in an orbit that is an ellipse, with the Sun at one focus of the ellipse.

![definitions](defs.png)

### Kepler's Second Law
The straight line joining a planet and the Sun sweeps out equal areas in space in equal intervals of time.

![equalareas](equalareas.png)

Kepler’s second law deals with the speed with which each planet moves along its ellipse, also known as its ***orbital speed***. Working with Brahe’s observations of Mars, Kepler discovered that the planet speeds up as it comes closer to the Sun and slows down as it pulls away from the Sun. He expressed the precise form of this relationship by imagining that the Sun and Mars are connected by a straight, elastic line. When Mars is closer to the Sun (positions 1 and 2 in Figure above), the elastic line is not stretched as much, and the planet moves rapidly. Farther from the Sun, as in positions 3 and 4, the line is stretched a lot, and the planet does not move so fast. As Mars travels in its elliptical orbit around the Sun, the elastic line sweeps out areas of the ellipse as it moves (the colored regions in our figure).In the figure above, the orbital speed of a planet traveling around the Sun (the circular object inside the ellipse) varies in such a way that in equal intervals of time (t), a line between the Sun and a planet sweeps out equal areas (A and B).

### Kepler's Third Law
The square of a planet’s orbital period is directly proportional to the cube of the semimajor axis of its orbit.

Kepler’s first two laws of planetary motion describe the shape of a planet’s orbit and allow us to calculate the speed of its motion at any point in the orbit. Kepler was pleased to have discovered such fundamental rules, but they did not satisfy his quest to fully understand planetary motions. He wanted to know why the orbits of the planets were spaced as they are and to find a mathematical pattern in their movements—a “harmony of the spheres” as he called it. For many years he worked to discover mathematical relationships governing planetary spacing and the time each planet took to go around the Sun.

In 1619, Kepler discovered a basic relationship to relate the planets’ orbits to their relative distances from the Sun. We define a planet’s orbital period, (P), as the time it takes a planet to travel once around the Sun. Also, recall that a planet’s semimajor axis, a, is equal to its average distance from the Sun. The relationship, now known as Kepler’s third law, says that a planet’s orbital period squared is proportional to the semimajor axis of its orbit cubed, or
$$P^2 \propto a^3$$
When 'P' (the orbital period) is measured in Earth-years, and 'a' is expressed in a quantity known as an astronomical unit (AU), the two sides of the formula are not only proportional but equal. In these "normalized" units (years and AU)
$$P^2 = a^3$$


![third](thirdlaw.png)

Note that one AU is the average distance between Earth and the Sun and is approximately equal to 150 million km. One Earth-year is approximately 365 days. Normalizing all the orbital parameters with respect to that of earth (ratios using Earth-based measurements set at 1), we arrive at the table below for the Solar System.

| Planet | Orbital Period 'P' years| Semi-major axis 'a' AU| Eccentricity of Orbit |
| ----------- | ----------- | ----------- | ----------- |
| Mercury | Title | Title | ----------- |
| Venus | Text | Text | ----------- |
| Earth | Title | Title | ----------- |
| Mars | Text | Text | ----------- |
| Jupiter | Title | Title | ----------- |
| Saturn | Text | Text | ----------- |
| Uranus | Title | Title | ----------- |
| Pluto | Text | Text | ----------- |
| Ceres | Title | Title | ----------- |
| Haumea | Text | Text | ----------- |
| Makemake | Title | Title | ----------- |
| Eris | Text | Text | ----------- |

In [4]:
# Read in the solar system planetary orbital parameters
import pandas as pd
data1 = pd.read_csv("updated_solar_system_orbits.csv")
data1

Planet  Semimajor Axis (AU)  Period (y)  Eccentricity  Mass of Planet
0    Mercury                 0.39        0.24         0.210             NaN
1      Venus                 0.72        0.60         0.010             NaN
2      Earth                 1.00        1.00         0.020             NaN
3       Mars                 1.52        1.88         0.090             NaN
4      Encke                 2.21        3.30         0.847             NaN
5      Vesta                 2.36        3.63         0.090             NaN
6      Ceres                 2.77        4.60         0.080             NaN
7     Pallas                 2.77        4.62         0.230             NaN
8    Jupiter                 5.20       11.86         0.050             NaN
9     Saturn                 9.54       29.46         0.060             NaN
10    Halley                17.94       76.10         0.967             NaN
11    Uranus                19.19       84.01         0.050             NaN
12   Neptune                30.06      164.82         0.010             NaN
13     Pluto                39.40      247.00         0.244             NaN
14    Haumea                43.18      283.77         0.194             NaN
15  Makemake                45.43      306.22         0.161             NaN
16      Eris                67.70      559.08         0.436             NaN

#### Let's check our understanding 
Use information in the table above as needed - answer these questions below by adding new cells for your calculations. Check your answers with the table above!
1. According to Kepler’s second law, where in a planet’s orbit would it be moving fastest? Where would it be moving slowest?
1. If the major axis of an elliptical orbit is 16 AU, what is the semimajor axis? If the eccentricity is 0.8, would this ellipse be best described as mostly circular or very elongated?
1. Mars takes 1.88 Earth years to go around the Sun. What is the semimajor axis of Mars from the Sun when measured in AU.
1. What is the semimajor axis (in astronomical units) of an asteroid with an orbital period of 8 years?
1. What is the average distance from the Sun (in astronomical units) of a planet with an orbital period of 45.66 years?
1. In 1996, astronomers discovered an icy object beyond Pluto that was given the designation 1996 TL 66. It has a semimajor axis of 84 AU. What is its orbital period according to Kepler’s third law?
1. Calculate 'Semifocal Distance' and 'Perihelion Distance' in AU for all of the planets above.